### PR TITLE
Allow kernel configuration via attributes

### PR DESCRIPTION
**Pull request by jdreesen**
This is a followup of #8 and improves it by adding attributes that allow to configure the kernel instead of passing a closure to `createKernel()`.

```php
use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureContainer;
use Neusta\Pimcore\TestingFramework\Test\Attribute\ConfigureExtension;
use Neusta\Pimcore\TestingFramework\Test\Attribute\RegisterBundle;
use Neusta\Pimcore\TestingFramework\Test\Attribute\RegisterCompilerPass;
use Neusta\Pimcore\TestingFramework\Test\ConfigurableKernelTestCase;

#[RegisterBundle(SomeBundle::class)]
class SomeTest extends ConfigurableKernelTestCase 
{
    #[ConfigureContainer(__DIR__ . '/Fixtures/some_config.yaml')]
    #[ConfigureExtension('some_extension', ['config' => 'values'])]
    #[RegisterCompilerPass(new SomeCompilerPass())]
    public function test_something(): void
    {
        self::bootKernel();

        // test something
    }
}
```

All attributes are usable on the class *and* test methods.

### CODE REVIEW

**Review by lukadschaak — APPROVED** [review]

File: src/Test/Reflection/TestAttributeProvider.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
+declare(strict_types=1);
+
+namespace Neusta\Pimcore\TestingFramework\Test\Reflection;
+
+use Neusta\Pimcore\TestingFramework\Test\Attribute\KernelConfiguration;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @internal
```
Comment (lukadschaak) [inline src/Test/Reflection/TestAttributeProvider.php:10]: Why internal? It is a test class and will not be available in vendor right?

Comment (jdreesen) [reply src/Test/Reflection/TestAttributeProvider.php:10]: No, it's in the `Test` namespace of the bundle and will/must be downloaded. This is a helper class that's needed to fetch the attributes from the class/methods.

Comment (lukadschaak) [reply src/Test/Reflection/TestAttributeProvider.php:10]: okay

**Review by yariksheptykin — COMMENTED** [review]
This is a great feature. Thanks!

**Pull request merged**